### PR TITLE
HDDS-3312. Remove server-side dependencies from hdds/ozone-common.

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

`hadoop-ozone/common` and `hadoop-hdds/common` projects are common between client and server. Therefore we should remove any server side utilities / dependencies from them as they would be added to the client classpath which makes harder the Ozone client / ozonefs adoption.

1. hadoop-hdds should depend on a minimal set of hadoop dependencies (can be managed a separated technical project / pom.xml)
2. code shared between server side projects (and not with the client) should be moved to the framework
3. OM related code should be moved to the ozone-manager instead of ozone/common
4. For technical reason some test utilities are imported to the Ozone code base (from Hadoop)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3312

## How was this patch tested?

Full integration test. This is a project organization patch without real change of the logic. If the clients can work without class-not-found exception. It should be fine.

### CODE REVIEW

File: hadoop-ozone/ozonefs-lib-current/pom.xml [hunk 1]
```diff
@@ -179,36 +179,36 @@
...
-        <exclusion>
-          <groupId>com.google.code.findbugs</groupId>
-          <artifactId>jsr305</artifactId>
-        </exclusion>
-      </exclusions>
+<!--      <exclusions>-->
+<!--        <exclusion>-->
+<!--          <groupId>org.apache.hadoop</groupId>-->
+<!--          <artifactId>hadoop-common</artifactId>-->
+<!--        </exclusion>-->
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs-lib-current/pom.xml:186]: These look strange.

File: hadoop-hdds/hadoop-dependency-client/README.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+This helper project provides a reduced view to the Hadoop dependencies. Hadoop common by default depends on a lot of other projects
+which are not required for the parts used in Ozone.
+
+The exclude rules should be updated on each version bump. If Hadoop introduce new dependencies we need to exclude them (unless they are required).
+
+
+For the reference here are the minimal jar files wich are required by Ozone:
+
+
+```
```
Comment (adoroszlai) [inline hadoop-hdds/hadoop-dependency-client/README.txt:27]: I think this file should be renamed to `README.md`.

File: hadoop-ozone/ozonefs-lib-current/pom.xml (hunk 1 above)
Comment (elek) [reply hadoop-ozone/ozonefs-lib-current/pom.xml]: Yes, I just noticed yesterday night. It's restored now.

**Comment by elek** [comment]
> It will have a few conflicts with #701 so if you need to update this anyway, I'd appreciate if you could review my PR. :)

@adoroszlai: #701 is already conflicting. Do a rebase and I will merge it.

**Comment by adoroszlai** [comment]
Thanks for merging #701 first.  In return I have merged `master` into this branch and resolved conflicts: https://github.com/adoroszlai/hadoop-ozone/commit/0a76d0cf0b02016962b4b32bb58f29fe4090f20f

Checkstyle still fails due to the new module `hadoop-hdds-hadoop-dependency-client`, fix is in progress in #768.

**Comment by adoroszlai** [comment]
Checkstyle problem is fixed on master.

Hadoop-related acceptance tests are failing:

```
hadoop31-hadoopo3fs :: Test ozone fs with hadoopfs                            
==============================================================================
Test hadoop dfs                                                       | FAIL |
1 != 0
```

due to

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/hadoop/ozone/shaded/org/apache/http/client/utils/URIBuilder
	at org.apache.hadoop.fs.ozone.BasicOzoneFileSystem.initialize(BasicOzoneFileSystem.java:141)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3303)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:124)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3352)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3320)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:479)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:361)
	at org.apache.hadoop.fs.shell.PathData.expandAsGlob(PathData.java:325)
	at org.apache.hadoop.fs.shell.CommandWithDestination.getRemoteDestination(CommandWithDestination.java:195)
	at org.apache.hadoop.fs.shell.CopyCommands$Put.processOptions(CopyCommands.java:259)
	at org.apache.hadoop.fs.shell.Command.run(Command.java:175)
	at org.apache.hadoop.fs.FsShell.run(FsShell.java:328)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:90)
	at org.apache.hadoop.fs.FsShell.main(FsShell.java:391)
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.ozone.shaded.org.apache.http.client.utils.URIBuilder
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
```

https://github.com/apache/hadoop-ozone/pull/744/checks?check_run_id=558258420
https://github.com/adoroszlai/hadoop-ozone/runs/559314792

**Comment by elek** [comment]
Merged to the master. Thanks the review (and the help with the rebase) @adoroszlai

**Pull request closed without merging**